File: Solutions/0409- Longest Palindrome/Solution.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int longestPalindrome(String s) {
        Set<Character> set = new HashSet<>();
        int count = 0;

        for (char ch : s.toCharArray()) {
            if (!set.contains(ch))
                set.add(ch);
            else {
                set.remove(ch);
                count += 2;
            }
        }



        return !set.isEmpty() ? count + 1 : count;
    }
}

File: Solutions/0409- Longest Palindrome/Solution.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        count = 0
        my_set = set()

        for ch in s:
            if ch not in my_set:
                my_set.add(ch)
            else:
                my_set.remove(ch)
                count += 2

        return count + 1 if my_set else count